### ISSUE TITLE
Is there possible add Quixote web framework support

### ISSUE THREAD

**Issue by ischaojie** [marshmallow-code/webargs#747]
My company heavily uses `Quixote` as a Web Framework, But now it lacks params validation and some others,

I found `webargs` this excellent lib for validating req obj, (Thanks for your job!)

So Is there possible add [Quixote](http://quixote.ca/) web framework support?

**Comment by ischaojie** [comment]
I notice `webargs` can use parser `parser.parse(user_args, request)`, I will try it first.

**Comment by sirosen** [comment]
We've moved away from adding built-in parsers for various frameworks, although there's no plan to remove the existing parsers and framework support. This has come up in the past for other frameworks too (e.g. #608).
So the short answer is "No", but it's not without its reasons and a (hopefully good) path for you to handle this case yourself.

To offer some of the rationale: supporting multiple frameworks within a single package has its challenges, and if we accepted new framework requests, we'd have to figure some kind of process for vetting and accepting or rejecting frameworks.

It is generally easy for users to define parsers for their framework of choice by inheriting from the base `webargs.Parser` class, and there are a number of third-party parser packages documented in the wiki on [the ecosystem page](https://github.com/marshmallow-code/webargs/wiki/Ecosystem).
If you or someone at your company were to write and publish a `webargs-quixote` package, we'd be more than happy to link it from the ecosystem page and readme here!

---

Okay, with that out of the way -- is it possible for you to write a `QuixoteParser(webargs.Parser)` class, and what would that look like?

Well... I'm not sure. I've never used Quixote, but I'm looking at the source and the demo dir right now. It seems like it might be difficult to handle.

The typical style for webargs parsers is that they offer the following methods:
- a method for getting the current request object (`flask.Request`, etc), called [`get_request_from_view_args`](https://webargs.readthedocs.io/en/latest/api.html#webargs.core.Parser.get_request_from_view_args)
- methods for getting dict-like data from the request object for various different sources, `load_headers`, `load_json`, etc.

and then the `webargs` decorators `use_args` and `use_kwargs` work as desired. The decorators need `get_request_from_view_args` in order to find the request object.

If it's hard to write `get_request_from_view_args`, I might recommend trying to get things working without the decorators, and only supporting explicit calls to [`Parser.parse`](https://webargs.readthedocs.io/en/latest/api.html#webargs.core.Parser.parse) at first. `parse` needs the request as its second argument, and it looks like `quixote.get_request` would serve that purpose.

The hardest part, the part where I would be most concerned about efforts to write a parser, is getting `load_json`, `load_headers`, `load_querystring`, `load_form`, and other methods implemented correctly. Looking at the quixote request object, it doesn't have built-in support for all of these types of data. I don't see JSON data support, or a way of getting all of the headers as a mapping (only `get_header` for an individual header), etc.
